cmd/up.go

```diff
@@ -379,6 +381,9 @@ func newUpCmd() *cobra.Command {
 		"target", "t", []string{},
 		"Specify a single resource URN to update. Other resources will not be updated."+
 			" Multiple resources can be specified using: --target urn1 --target urn2")
+	cmd.PersistentFlags().BoolVar(
+		&targetDependents, "target-dependents", false,
+		"Allows updating of dependent targets discovered but not specified in --target list")
```

Comment:
I still think that this naming is misleading. Resources that are being created but are not in the `--target` are orthogonal to changes to dependent resources.

Reply:
That said, I don't have a better suggestion, and I do think that we could investigate the real version of what this means, i.e. allowing changes to flow downstream from targeted resources to untargeted resources.

Reply:
this is why i had this be "force targets" initially.  because it's basically the way for users to say "yes, i'm forcing you to do whatever you want with resources that i didn't explicitly state in the --xxx-targets list.  It's basically the 'escape hatch' to prevent people from having to state every resource that we think needs to be touched.

Reply:
Yeah, there's just something about the name "force targets" that really rubs me wrong. IMO it doesn't really give you any info about what it does.

Reply:
`--allow-unmentioned-targets`? `allow-unspecified-targets`?

Reply:
I think what's actually nagging at me here is that our behavior for creates just feels wrong overall.

For a resource that already exists, we simply treat it as unchanged if it's not mentioned in the targets list. The equivalent of this for a create would be to _not create the resource at all_. Instead, we issue an error informing the user that the resource _would_ be created, but won't be because it's not in the targets list. What if we stopped doing that?

The obvious downside to not mentioning resources that are not being created because they are not targeted is that it is harder for the user to do exactly a targeted create, because they have to guess the URN. So we probably still want to emit some message. I don't think that allowing all untargeted creates using a flag is actually the right thing to do, though.

Reply:
> Instead, we issue an error informing the user that the resource would be created, but won't be because it's not in the targets list. What if we stopped doing that?

What are the implications here then?  Say if another existing resource depended on the output of a resource that is not being created?

Reply:
It would see an unknown input during the preview and an undefined value during the update. If the user needs the resource that is not being created to be created in order to perform the update, then they'll have to specify it as a target.

Reply:
> If the user needs the resource that is not being created to be created in order to perform the update, then they'll have to specify it as a target.

but isn't that precisely what thsi change is telling them. it's very much a: 

1. this seems like you're doing somethin really wonky.  we'll error to let you know about it.
2. you can get through by being explicit, or here's an escape hatch .

"1" is to prevent random unexpected strangeness that has wandered into the territory of "this most likely seems like we shouldn't proceed".
"2" is to allow users a gentle path to deal with it.  i.e. instead of forcing them to have to specify all the urns, they can just go "yes, i get it, just proceed".  The latter may be nice as perhaps their program does something that would then cause them to have to painfully name a whole host of resources.

i'm trying to find the right balance of protection, power and flexiblity for the user.  The approach where we just ignore the creates, but then might leave random `undefine`s flowing through just doesn't actually seem to buy people anything.  It seems like it would be a worse situation overall versus being conservative by default and just asking users to confirm what they want with an affirmative step on their part.

Reply:
Another way to put this: i think if i needed to explain to users the behavior of 'update' with a new resource being created, and how that would impact downstream resources, i think it would be too confusing and too error prone.  I think it's much saner and easier to understand as simply: if we detect that your program would create or destroy a resource, we're going to error if you didn't specify it.  

Note that because of our difficulty defining or accepting 2.2, i'm leaning just toward 2.1.  We tell you there's an issue and your workaround is to just be explicit on the command line.